include/nbl/video/IPhysicalDevice.h

```diff
@@ -512,11 +512,42 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             }
         }
 
+        
+        static inline uint32_t getMaxInvocationsPerComputeUnitsFromDriverID(E_DRIVER_ID driverID)
+        {
+            const bool isIntelGPU = (driverID == E_DRIVER_ID::EDI_INTEL_OPEN_SOURCE_MESA || driverID == E_DRIVER_ID::EDI_INTEL_PROPRIETARY_WINDOWS);
+            const bool isAMDGPU = (driverID == E_DRIVER_ID::EDI_AMD_OPEN_SOURCE || driverID == E_DRIVER_ID::EDI_AMD_PROPRIETARY);
+            const bool isNVIDIAGPU = (driverID == E_DRIVER_ID::EDI_NVIDIA_PROPRIETARY);
+            if (isNVIDIAGPU)
+                return 32u * 48u; // RTX 3090 (32 Threads/Warp * 48 Warp/SM)
+            else if (isAMDGPU)
+                return 64u * 1024u; // RX 6900XT (64 Threads/Wave * 1024 Threads/Wave) https://gpuopen.com/learn/optimizing-gpu-occupancy-resource-usage-large-thread-groups/
```

Comment:
don't RDNA 1&" have 32 threads per wave now?

Reply:
Not sure about the 64 or 32, 
I saw some comment on Wikipedia that our supports both wavefront sizes so I chose the max
https://en.m.wikipedia.org/wiki/RDNA_(microarchitecture)#:~:text=RDNA%20supports%20both%20wavefront%20sizes%20of%2032%20and%2064%20threads.
Your call

Reply:
> Not sure about the 64 or 32, I saw some comment on Wikipedia that our supports both wavefront sizes so I chose the max https://en.m.wikipedia.org/wiki/RDNA_(microarchitecture)#:~:text=RDNA%20supports%20both%20wavefront%20sizes%20of%2032%20and%2064%20threads. Your call

I'd guess that switching to a higher wavefront size, doesn't magically increase the addressable register file or smem size by 100%, so the overall occupancy limit stays constant.